ui/intermittent-failures/App.jsx

```diff
@@ -1,31 +1,54 @@
 import React from 'react';
 import { HashRouter, Route, Switch, Redirect } from 'react-router-dom';
-import PropTypes from 'prop-types';
-import { Provider } from 'react-redux';
 
 import MainView from './MainView';
 import BugDetailsView from './BugDetailsView';
 
-function App({ store }) {
-  return (
-    <Provider store={store}>
+export default class App extends React.Component {
+    constructor(props) {
+    super(props);
+    this.updateAppState = this.updateAppState.bind(this);
+
+    // keep track of the mainviews graph data so the API won't be
+    // called again when navigating back from bugdetailsview;
+    // table API will be called every time it mounts.
+    this.state = { graphData: null };
+  }
+
+  updateAppState(state) {
+    this.setState(state);
+  }
+
+  render() {
+    return (
       <HashRouter>
         <main>
           <Switch>
-            <Route exact path="/main" component={MainView} />
-            <Route path="/main?startday=:startday&endday=:endday&tree=:tree" component={MainView} />
+            (<Route
+              exact
+              path="/main"
+              render={props =>
+              (<MainView
+                {...props}
+                mainGraphData={this.state.graphData}
+                updateAppState={this.updateAppState}
+              />)}
+            />)
+            (<Route
+              path="/main?startday=:startday&endday=:endday&tree=:tree"
+              render={props =>
+                (<MainView
+                  {...props}
+                  mainGraphData={this.state.graphData}
+                  updateAppState={this.updateAppState}
+                />)}
+            />)
             <Route path="/bugdetails" component={BugDetailsView} />
             <Route path="/bugdetails?startday=:startday&endday=:endday&tree=:tree&bug=bug" component={BugDetailsView} />
             <Redirect from="/" to="/main" />
           </Switch>
         </main>
       </HashRouter>
-    </Provider>
-  );
+    );
+  }
 }
-
-App.propTypes = {
-  store: PropTypes.shape({}).isRequired,
-};
-
-export default App;
```

Comment:
Could you keep the `export` separate, since in my Neutrino 9 PR I need to wrap `App` for the new react-hot-loader v4 way of configuring the app.

Reply:
Sure, will do.